Move AI reformatting plugin to submodule

andybroomfield opened the pull request:
Fix #16

<!-- See https://docs.localgovdrupal.org/contributing/ for guidelines on contributing. -->
## What does this change?

- Move the AI plugin into a submodule
- Move the drupal/ai dependency to composer suggest
- Update readme (AI submodule references)

<!-- A pull request should have enough detail to be understandable far in the
future. e.g what is the problem/why is the change needed, how does it solve it
and any questions or points of discussion. Prefer copying information from a
Trello card (for example) over linking to it; the card may not always exist and
reviewers may not have access to the board. -->

## How to test

- Module should now be installable without the AI module needing to be enabled, and a PDF (with very basic formatting) can be imported.
- Running `composer suggest` should indicate that the module can be enhanced with the `drupal/ai` module.
- Enabling the submodule requires the Drupal/AI module to be present, and then once set up with at least one provider then a PDF should be imported with some formatting.
- Check that the updated README.md and composer suggestions make sense.

<!-- Provide instructions to help others verify the change. This could take the
form of "On main, do X and witness Y. On this branch, do X and witness Z. " -->

## How can we measure success?

More people can install the importer, and those that do don't require just Open AI to be enabled as a provider.

<!-- Do you expect errors to decrease? Do you expect user journeys to be
simplified? What can be used to prove this? A filtered view of logs or
analytics, etc? -->

## Have we considered potential risks?

<!-- What are the potential risks and how can they be mitigated? Does an error
require an alarm? Should user help, infosec, or legal be informed of this
change? Is private information guarded? Do we need to add anything in the
backlog? -->

How this gels with changes in #15. That one possibly needs to merge first and then this PR rebased.

## Images

n/a
<!-- Usually only applicable to UI changes, what did it look like before and
what will it look like after? -->

## Accessibility

n/a

<!-- Usually only applicable to UI changes, check the boxes if you are satisfied that your changes pass these tests -->

andybroomfield committed:
Move the AI plugin into a submodule

- Creates localgov_publications_importer_ai submodule
- Move the AI transform plugin into this module
- Remove AI as a dependency on the main module

andybroomfield committed:
Move the drupal/ai dependency to composer suggest

andybroomfield committed:
Update readme (AI submodule references)

Add information about the new AI submodule and indicate that this module and an AI
provider needs to be installed in order to use AI to reformat the PDF.

andybroomfield committed:
Amend composer suggests text

Add that drupal/ai is required by the localgov_publications_importer_ai submodule.

andybroomfield committed:
Remove debug lines

andybroomfield committed:
Move position of use statement to correct order

rupertj commented on README.md:
```diff
@@ -16,17 +16,19 @@ You can fund the development of this feature via the [LocalGov Drupal Community
-7) Scroll down to chat. Ensure OpenAI is selected. Choose the model you'd like to use. GPT-4o seems to work.
+## Using AI to format the imported PDF
+
+If you'd like to use AI to clean up the text, you can. A submodule, localgov_publications_importer_ai, is included. To enable this module you will need to install the [Drupal AI module](https://www.drupal.org/project/ai) and at least one AI provider module. The default AI chat provider will be used if one is configured. To configure one using ChatGPT, you'll need to get an API key from OpenAI, th ...
```
I think this could make it a bit clearer than you can use AI's other than ChatGPT. Maybe mentioning at least one other might help?

rupertj reviewed (approved):
One minor comment on the docs, but otherwise this looks good to me.

andybroomfield committed:
Update readme with other AI provider references

In official BBC style voice: "Other AI LLM providers are also available."
The link goes to the AI module page where it lists other provider modules.
Also documents where to get an Open AI api key from.

andybroomfield commented:
Updated readme @rupertj  https://github.com/localgovdrupal/localgov_publications_importer/blob/a805b93c33b2775141ad11561cf8ba475d34230e/README.md

andybroomfield committed:
Remove Open AI php library dependency

This will be brought in if a user requires drupal/ai_provider_open_ai

finnlewis commented:
Discussing in Merge Tuesday, @rupertj is happy, @andybroomfield  is happy.
